doc/general_design.md

```diff
@@ -4,31 +4,79 @@
 
 > Implement a prototype job worker service that provides an API to run arbitrary Linux processes.
 
-## General Design ##
+## Design ##
```

Comment:
Note: I would have found it much easier for the PR to contain the _entire_ design document as "new" code. The diffs here make it harder to follow.